### PR TITLE
feat: add likes and titles to projects and reports

### PR DESCRIPTION
**Pull request by Sparkier**
# Description

fix ZEN-268

image
image

### CODE REVIEW

**Comment by cabreraalex** [comment]
I think the description should be inline with the project name in Projects? Cut off if too long with ellipses, and we can show the full thing in a tooltip on hover.

**Comment by Sparkier** [comment]
> I think the description should be inline with the project name in Projects? Cut off if too long with ellipses, and we can show the full thing in a tooltip on hover.

Kk can change tomorrow don't have real preferences here.

File: frontend/src/routes/(app)/report/[owner]/[report]/+page.svelte [hunk 1]
```diff
@@ -82,6 +85,13 @@
 			<h1 class="text-5xl mr-6 text-grey-darkest">
 				{data.report.name}
 			</h1>
+			<LikeButton
+				on:like={() => zenoClient.likeReport(data.report.id)}
```
Comment (cabreraalex) [inline frontend/src/routes/(app)/report/[owner]/[report]/+page.svelte:89]: Can this happen inside LikeButton?

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: backend/zeno_backend/database/select.py [hunk 2]
```diff
@@ -580,6 +596,8 @@ def report(
                     element_result,
                 )
             ),
+            num_likes=num_likes[0][0] if isinstance(num_likes[0][0], int) else 0,
```
[the remark this answers is not in the document]
Comment (Sparkier) [reply backend/zeno_backend/database/select.py:599]: If we want to do this should do everywhere. Right now we check int types throughout. Won't change for this PR.

[the remark this answers is not in the document]
Comment (Sparkier) [reply backend/zeno_backend/database/select.py:599]: You did so for the other like stuff as well btw, just copied the code from there.

File: frontend/src/routes/(app)/report/[owner]/[report]/+page.svelte (hunk 1 above)
Comment (Sparkier) [reply frontend/src/routes/(app)/report/[owner]/[report]/+page.svelte:89]: Can pass a function or do this. I think this is cleaner tbh.

File: frontend/src/lib/components/general/LikeButton.svelte [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+					liked = true;
+					likes++;
+				}
+				dispatch('like');
+			}}
+		>
+			<Tooltip
+				content={`Like this ${report ? 'report' : 'project'}!`}
+				theme={'zeno-tooltip'}
+				position="bottom"
```
[the remark this answers is not in the document]
Comment (Sparkier) [reply frontend/src/lib/components/general/LikeButton.svelte:38]: Good catch. Need to move the like button next to the title with our tooltip now. I think it's fine, though.

File: frontend/src/lib/components/project/ProjectHeader.svelte [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	export let project: Project;
+	export let likes: number;
+	export let liked: boolean;
+	export let user: User | null;
+
+	const zenoClient = getContext('zenoClient') as ZenoService;
+</script>
+
+<div class="flex p-4 bg-yellowish-light border-b border-b-grey-lighter items-center min-w-0 w-full">
+	<h1 class="text-xl font-semibold">{project.name}</h1>
```
Comment (cabreraalex) [inline frontend/src/lib/components/project/ProjectHeader.svelte:15]: add `shrink-0` so it doesn't wrap when description is long

File: frontend/src/lib/components/project/ProjectHeader.svelte [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	<div class="pl-4">
+		<LikeButton on:like={() => zenoClient.likeProject(project.uuid)} {likes} {liked} {user} />
+	</div>
+	<p class="pl-4 text-grey-dark text-ellipsis overflow-hidden whitespace-nowrap">
```
Comment (cabreraalex) [inline frontend/src/lib/components/project/ProjectHeader.svelte:19]: I think I like it more with the like button on the top right always IMO.

Comment (cabreraalex) [inline frontend/src/lib/components/project/ProjectHeader.svelte:19]: `py-3 px-5`?

File: frontend/src/lib/components/general/LikeButton.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (cabreraalex) [reply frontend/src/lib/components/general/LikeButton.svelte:38]: re: my other comment on top right like, can just make it a left tooltip

[the remark this answers is not in the document]
Comment (cabreraalex) [reply frontend/src/lib/components/general/LikeButton.svelte:38]: probably make it a prop

**Comment by cabreraalex** [comment]
> add `shrink-0` so it doesn't wrap when description is long 

comment didn't save but also can do `font-header` can be fun.

File: frontend/src/lib/components/project/ProjectHeader.svelte (hunk 5 above)
Comment (Sparkier) [reply frontend/src/lib/components/project/ProjectHeader.svelte:19]: tried px-5 looks kinda off with the other stuff

Comment (cabreraalex) [reply frontend/src/lib/components/project/ProjectHeader.svelte:19]: yeah no worries other was fine

File: frontend/src/lib/components/project/ProjectHeader.svelte (hunk 4 above)
Comment (Sparkier) [reply frontend/src/lib/components/project/ProjectHeader.svelte:15]: don't know, personally think it looks off

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**